### PR TITLE
Prestashop plugin vue

### PR DESCRIPTION
**Pull request by vegimcarkaxhija**

### CODE REVIEW

File: library/checkout/in3checkout.php [hunk 1]
```diff
@@ -158,21 +159,21 @@ protected function getBillingAddress()
     {
         $data = [
             'recipient' => [
-                'category'              => 'B2C',
-                'initials'              => initials($this->invoice_address->firstname),
-                'firstName'             => $this->invoice_address->firstname,
-                'lastName'              => $this->invoice_address->lastname,
-                'birthDate'             => date(
+                'category' => 'B2C',
+                'initials' => initials($this->invoice_address->firstname),
```
Comment (harli91) [inline library/checkout/in3checkout.php:163]: this is a global function, may conflict with other functions named like this

File: src/Repository/ConfigurationRepository.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        return true;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function getActiveCreditCards(): array
+    {
+        $paymentName = 'creditcard';
```
Comment (harli91) [inline src/Repository/ConfigurationRepository.php:64]: why do you need this variable?

File: src/Repository/CountryRepository.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        return $result;
+    }
+
+    public function checkAndInsertNewCountries()
+    {
+        $dbCountries = $this->getCountriesFromDB();
+        $dbCountryIds = array_column($dbCountries, 'id');
+
+        $langId = \Context::getContext()->language->id;
+        $rawCountries = \Country::getCountries($langId, true);
```
Comment (harli91) [inline src/Repository/CountryRepository.php:68]: if the countries are in prestashop why do you need to create a new table for countries??

File: src/Repository/OrderingRepository.php [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        }
+
+        $ordering->setValue($value);
+        $this->_em->persist($ordering);
+        $this->_em->flush();
+
+        return true;
+    }
+
+    private function updateOrderingForNullCountry($value)
```
Comment (harli91) [inline src/Repository/OrderingRepository.php:94]: these code seems to repeat with the code from the method above, remove the duplicate code

File: src/Repository/OrderingRepository.php [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+            'status' => true,
+        ];
+
+        $paymentMethodIds = is_string($ordering->getValue())
+            ? json_decode($ordering->getValue(), true)
+            : $ordering->getValue();
+
+        foreach ($paymentMethodIds as $id) {
+            $paymentMethodRepo = $this->_em->getRepository(BkPaymentMethods::class);
+            $paymentMethodData = $paymentMethodRepo->findOneById($id);
```
Comment (harli91) [inline src/Repository/OrderingRepository.php:133]: why are you doing so many database calls? why not retrieve all the payment methods first?

File: src/Service/BuckarooSettingsService.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    public function updateSettings($data)
+    {
+        \Configuration::updateValue(Config::BUCKAROO_MERCHANT_KEY, $data['website_key']);
+        \Configuration::updateValue(Config::BUCKAROO_SECRET_KEY, $data['secret_key']);
+        \Configuration::updateValue(Config::BUCKAROO_TRANSACTION_LABEL, $data['transaction_description']);
+        \Configuration::updateValue(Config::BUCKAROO_TEST, $data['is_live']);
+        \Configuration::updateValue(Config::LABEL_REFUND_RESTOCK, $data['restock']);
+        \Configuration::updateValue(Config::LABEL_REFUND_CREDIT_SLIP, $data['creditSlip']);
+        \Configuration::updateValue(Config::LABEL_REFUND_VOUCHER, $data['voucher']);
+        \Configuration::updateValue(Config::LABEL_REFUND_CREATE_NEGATIVE_PAYMENT, $data['negativePayment']);
```
Comment (harli91) [inline src/Service/BuckarooSettingsService.php:61]: I think you have a method to update multiple config values at once, can reduce the database queries

File: upgrade/upgrade-4.0.1.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+ *  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+/**
+ * @return mixed
+ */
+function upgrade_module_4_0_1($object)
```
Comment (harli91) [inline upgrade/upgrade-4.0.1.php:24]: the latest release is 4.0.0,I don't think this migration will run

File: controllers/front/creditcards.php [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+        $this->creditCardsRepository = new RawCreditCardsRepository();
+    }
+
+    public function initContent()
+    {
+        parent::initContent();
+        $this->authenticate();
+
+        $countries = $this->creditCardsRepository->getCreditCardsFromDB();
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply controllers/front/creditcards.php:38]: we have an option to activate and deactivate specific credit cards, and I used ids of creditcard table for that

File: controllers/front/request.php [hunk 9]
```diff
@@ -112,11 +121,19 @@ public function postProcess()
         $logger->logInfo('Checkout info', $debug);
 
         $this->checkout = Checkout::getInstance($payment_method, $cart);
+        $this->checkout->platformName = 'PrestaShop';
+        $this->checkout->platformVersion = _PS_VERSION_;
+        $this->checkout->moduleSupplier = $this->module->author;
+        $this->checkout->moduleName = $this->module->name;
+        $this->checkout->moduleVersion = $this->module->version;
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply controllers/front/request.php:128]: not fixed

File: dev/src/components/PaymentMethodBlock.vue [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+                }
+            })
+            .catch((e) => {
+                this.toastr.error(`Something went wrong...`)
+            })
+        },
+    },
+    setup(props) {
+        const { toastr } = useToastr()
+        const { post, data, loading, setEndpoint } = useApi('/index.php?fc=module&module=buckaroo3&controller=paymentMethodMode')
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply dev/src/components/PaymentMethodBlock.vue:56]: not fixed

File: dev/src/components/VerificationMethodBlock.vue [hunk 11]
```diff
@@ -0,0 +1,71 @@
+<template>
+    <div class="bg-gray-50 p-3 rounded">
+        <div class="text-center space-y-2">
+            <div class="flex justify-center">
+                <img :src="`/modules/buckaroo3/views/img/buckaroo/Identification methods/SVG/${ payment.icon }`" alt="icon" class="w-12">
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply dev/src/components/VerificationMethodBlock.vue:5]: not fixed

File: src/Repository/CountryRepository.php (hunk 3 above)
Comment (vegimcarkaxhija) [reply src/Repository/CountryRepository.php:68]: I use id`s of saved countries in ordering

File: src/Service/BuckarooSettingsService.php (hunk 6 above)
Comment (vegimcarkaxhija) [reply src/Service/BuckarooSettingsService.php:61]: can you show me that method, I couldn't find it

File: upgrade/upgrade-4.0.1.php (hunk 7 above)
Comment (vegimcarkaxhija) [reply upgrade/upgrade-4.0.1.php:24]: no latest one is 4.0.1

**Review by harli91 — APPROVED** [review]

**Pull request merged**